src/MongoDB/Manager.c

```diff
@@ -279,7 +279,9 @@ PHP_METHOD(Manager, selectServer)
 	}
 
 	readPreference = phongo_read_preference_from_zval(zreadPreference TSRMLS_CC);
-	selected_server = mongoc_topology_select(intern->client->topology, MONGOC_SS_READ, readPreference, MONGOC_SS_DEFAULT_LOCAL_THRESHOLD_MS, &error);
+	/* TODO: Remove (mongoc_read_prefs_t *) casting once
+	 * https://github.com/mongodb/mongo-c-driver/pull/315 is merged. */
+	selected_server = mongoc_client_select_server(intern->client, true, (mongoc_read_prefs_t *) readPreference, &error);
```

Comment:
My mistake. I should have been passing `false` for `for_writes`. Aside from the error checking (`for_writes=true` requires that no read prefs are passed`), this ultimately gets used in `mongoc_topology_description_suitable_servers()` and merely determines whether read prefs get applied for selection. And when a writable server is required, it can skip read pref checking (and also not require a primary-only read pref).